pandas/io/common.py

```diff
@@ -184,13 +185,25 @@ def is_gcs_url(url) -> bool:
 
 
 def urlopen(*args, **kwargs):
-    """
-    Lazy-import wrapper for stdlib urlopen, as that imports a big chunk of
-    the stdlib.
-    """
-    import urllib.request
+    compression = None
+    content_encoding = None
+    try:
+        requests = import_optional_dependency("requests")
+        r = requests.get(*args, **kwargs)
+        r.raise_for_status()
+        content = r.content
+        r.close()
+    except ImportError:
+        import urllib.request
 
-    return urllib.request.urlopen(*args, **kwargs)
+        r = urllib.request.urlopen(*args, **kwargs)
+        content = r.read()
+        content_encoding = r.headers.get("Content-Encoding", None)
+    if content_encoding == "gzip":
```

Comment:
I think this also needs to be under the `except ImportError`? From what I can tell, requests `.content` will automatically decode gzip: https://requests.readthedocs.io/en/master/user/quickstart/#binary-response-content